integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/script/JsonSupportByScriptMediatorTestCase.java

```diff
@@ -0,0 +1,144 @@
+                httpHeaders);
+        boolean propertySet;
+        assertNotNull(response, "Response message null");
+        propertySet = isPropertyContainedInLog("JSON_TEXT = {\"name\":\"John Doe\",\"address\":\"California\","
+                + "\"dob\":\"1990-03-19\"}");
+        Assert.assertTrue(propertySet, " The serialized json payload is not set as a property ");
+
+    }
+
+    @Test(groups = {"wso2.esb"}, description = "Parsing serialized JSON payload with JS")
+    public void testParsingSerializedJson() throws Exception {
+
+        Map<String, String> httpHeaders = new HashMap<>();
+        httpHeaders.put("Content-Type", "application/json");
+        String payload = "{\n" + "\"name\": \"John Doe\",\n" + "\"dob\": \"1990-03-19\",\n" + "\"ssn\": "
+                + "\"234-23" + "-525\",\n" + "\"address\": \"California\",\n" + "\"phone\": \"8770586755\",\n"
+                + "\"email\":" + " \"johndoe@gmail.com\",\n" + "\"doctor\": \"thomas collins\",\n" + "\"hospital\": "
+                + "\"grand oak " + "community hospital\",\n" + "\"cardNo\": \"7844481124110331\",\n"
+                + "\"appointment_date\": " + "\"2017-04-02\"\n" + "}";
+        HttpResponse response = doPost(new URL(getApiInvocationURL("scriptMediatorJsParse")), payload,
+                httpHeaders);
+        Assert.assertTrue((response.getData().contains("California")), "Response does not contain "
+                + "the keyword \"California\". Response: " + response.getData());
+
+    }
+
+    @Test(groups = {"wso2.esb"}, description = "Handling null JSON objects with JS")
+    public void testHandlingNullJsonObjects() throws Exception {
+
+        Map<String, String> httpHeaders = new HashMap<>();
+        httpHeaders.put("Content-Type", "application/json");
+        String payload = "{\n" + "\"name\": \"John Doe\",\n" + "\"dob\": \"1990-03-19\",\n" + "\"ssn\": "
+                + "\"234-23" + "-525\",\n" + "\"address\": \"California\",\n" + "\"phone\": \"8770586755\",\n"
+                + "\"email\":" + " \"johndoe@gmail.com\",\n" + "\"doctor\": \"thomas collins\",\n" + "\"hospital\": "
+                + "\"grand oak " + "community hospital\",\n" + "\"cardNo\": \"7844481124110331\",\n"
+                + "\"appointment_date\": " + "\"2017-04-02\"\n" + "}";
+        HttpResponse response = doPost(new URL(getApiInvocationURL("scriptMediatorJsHandlingNullJsonObject")), payload,
+                httpHeaders);
+        Assert.assertTrue((response.getData().contains("{}")), "Response does not contain "
+                + "the keyword \"{}\". Response: " + response.getData());
+
+    }
+
+    /**
+     * This method check whether given property contains in the logs.
+     * 
+     * @param property required property which needs to be validate if exists or not.
+     * @return A Boolean
+     */
+    private boolean isPropertyContainedInLog(String property) throws LogViewerLogViewerException, RemoteException {
```

Comment:
Better to move this private method to the end of the class, giving priority to public methods.